Merge sections with identical title

krlmlr opened the pull request:
- Add test
- In order of appearence (it would be better to use the contents of the "main" file first)

Fixes #300.

hadley commented on R/rd-tag-api.R:
```diff
@@ -53,6 +53,29 @@ merge.minidesc_tag <- function(x, y, ...) {
 
+#' @export
+merge.section_tag <- function(x, y, ...) {
+  x_names <- sapply(x$values, `[[`, "name")
```
Could you please use `vapply()` here?

hadley commented:
Could you please also update news and squash?

krlmlr commented:
Now using vapply() instead of sapply() (there are still two instances of sapply() in the code, in process_def_tag()). Moved and changed NEWS.

Why are squashed pull requests important to you? I've only recently discovered `git log --first-parent`, which does the "right thing" for the master branch. See also http://devblog.nestoria.com/post/98892582763/maintaining-a-consistent-linear-history-for-git.

hadley commented:
It's not super important, but there are 16 commits here for a relatively small change - it makes understanding the history that much harder if I look at it in the future

krlmlr commented:
The history is an issue, but this is what `--first-parent` is for. Merged pull requests are shown as a single unit, without the detailed commits. (Also works with `gitk`, but then I don't know which tool you use to browse history.)

I'll squash the commits in a minute.

krlmlr commented:
Removed and squashed irrelevant commits.

hadley commented:
Thanks!
